### ISSUE TITLE
Refreshing Hapi static site in safari intermittently results in blank page

### ISSUE THREAD

**Issue by inator** [hapijs/hapi#2676]
I have a site with static html content that is SSL reverse proxied to a Node.js HAPI backend. The site renders fine in Chrome and Firefox, but when refreshing the site in Safari Version 8.0.7 (10600.7.12), a blank page is rendered with no error or sign of trouble other than this output:

    <html><head></head><body></body></html>

I have absolutely no problems otherwise, and the issue appears randomly and very occasionally. There are no console errors, but the response code is 304 (cached) - but with only the above DOM Tree content. Having not turned up any additional helpful info through research, I turn to the community for help:

Any ideas on what the problem may be?

Any suggestions on how to troubleshoot this?

Here's some response header info:

    Vary    origin
    Strict-Transport-Security   max-age=15768000
    Access-Control-Allow-Methods    GET, HEAD, POST, PUT, PATCH, DELETE, OPTIONS
    Content-Type    text/html; charset=utf-8
    Access-Control-Expose-Headers   WWW-Authenticate, Server-Authorization
    Cache-Control   no-cache
    Access-Control-Allow-Credentials    true
    Access-Control-Max-Age  86400
    Access-Control-Allow-Headers    Authorization, Content-Type, If-None-Match

**Comment by kanongil** [comment]
Hmm, this might be related to the bug reported in #2673.

The Safari problem is probable due to some kind of race condition. I'm guessing that the `Cache-Control: no-cache` header causes Safari to evict the cached response, and that occasionally this happens before it is presented, leading Safari to output a default blank html page (which matches the output you specify).

**Comment by inator** [comment]
@kanongil Thanks!  Any suggestion on how I might prevent this behavior as a stop gap until the bug is fixed?

**Comment by kanongil** [comment]
Yes, there is a simple work-around. Just add the 304 code to the cacheable responses using the `cache. statuses` connection option. Eg.:

```js
var server = new Hapi.Server();
server.connection({ …, cache: { statuses: [200, 304] } });
```

Please let me know if this actually solves the issue.

**Comment by inator** [comment]
@kanongil Unfortunately, right now I have to run Hapi 6.4.0 (will port to newer eventually), and the multiple servers are established via the Pack method:

    var pack = new Hapi.Pack();

Can you suggest a work around within this context?

**Comment by kanongil** [comment]
You really should update, as it is no longer supported.

**Comment by inator** [comment]
@kanongil Sorry - The 6.4.0 version is included in a larger framework that I don't have direct control over, but I know it'll be ported soon.  It was only after some investigation that I realized your work around wouldn't work due to the serious lag in HAPI version used.  Knowing that the port will likely be a couple of months out, I'm hoping you might have a suggestion I could try in the mean time.  If not, I understand. :)

**Comment by hueniverse** [comment]
I am going to close this since version 6.x is not supported anymore. Please do open a new issue once you upgrade if this problem still exist. Sorry we can't be more helpful.

**Comment by inator** [comment]
Just a follow-up.  This does not appear to be an issue with Safari 9.0. :)
